### PR TITLE
feat: add plugin interface

### PR DESCRIPTION
**Pull request by andrewrynhard**
This moves the render functionality to a plugin. This will allow for multiple implementations of render to exist (e.g. Helm based, Kustomize based, go template based).

Building the default render plugin looks like:

```
cd cmd/render
go build -buildmode=plugin -o render.so .
```

Then you invoke `bootkube` like so:

```
cd cmd/bootkube
go build .
bootkube render --asset-dir=./assets --plugin ../render/render.so --plugin-flag='--service-cidr=10.0.0.1/23,--cluster-name=example'
```

Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

**Comment by andrewrynhard** [comment]
@rmenn When you get a change PTAL. There is a bug in `master` that causes render to panic, see #1082, but I think this is good enough for a first pass at a plugin mechanism for rendering.

File: pkg/recovery/etcd.go [hunk 1]
```diff
@@ -10,9 +10,8 @@ import (
 	"path"
 	"strings"
 
-	"github.com/kubernetes-sigs/bootkube/pkg/asset"
-
 	"github.com/coreos/etcd/clientv3"
+	"github.com/kubernetes-sigs/bootkube/cmd/render/plugin/default/asset"
```
Comment (invidian) [inline pkg/recovery/etcd.go:14]: I think this import separation is fine, this is what `goimports` enforces, to keep imports within the module separate from external dependencies.

Usually you have 3 groups: stdlib, external, internal, in this order.

File: pkg/plugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package plugin
+
+type Options struct {
```
[the remark this answers is not in the document]
Comment (rmenn) [reply pkg/plugin/plugin.go:3]: I believe docs, build & makefile changes are pending. I think andrew just wanted to get a sign off on the implementation

[the remark this answers is not in the document]
Comment (andrewrynhard) [reply pkg/plugin/plugin.go:3]: Yeah this is going to be an iterative process I think. We have a lot of work to get bootkube back into shape even outside of this change. I can at least add golang comments.

File: pkg/plugin/plugin.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+	AssetDir string
+}
+
+type Renderer interface {
+	Render(*Options, []string)
```
[the remark this answers is not in the document]
Comment (andrewrynhard) [reply pkg/plugin/plugin.go:8]: Absolutely.

File: pkg/recovery/etcd.go (hunk 1 above)
Comment (andrewrynhard) [reply pkg/recovery/etcd.go]: I usually have `gofumports` to handle that ;) . Will fix 👍.

**Comment by andrewrynhard** [comment]
@invidian Do you think we could get the helm stuff working using this?

**Comment by andrewrynhard** [comment]
> > Then you invoke bootkube like so:
> 
> I think it would be nice to have `default` plugins built-in, so user does not need 2 files or specify the parameter explicitly to use the default plugins.
> 
> I like the PR in general, it seems like a step in good direction. I left some comments I would change before merging.

We had some thoughts about moving render out of bootkube entirely, and having dedicated repos eventually. Then @rmenn had the idea of downloading the default plugin on the fly. Still rough ideas.

**Comment by invidian** [comment]
>@invidian Do you think we could get the helm stuff working using this?

I guess so, if we define `Installer()` interface, it should be fine.

**Comment by andrewrynhard** [comment]
> > @invidian Do you think we could get the helm stuff working using this?
> 
> I guess so, if we define `Installer()` interface, it should be fine.

Great. Once this is merged I can start working on that interface as well.

**Pull request merged**